src/Builder/Encoder/OperatorEncoder.php

```diff
@@ -129,6 +132,34 @@ private function encodeAsObject(OperatorInterface $value): stdClass
             : $this->wrap($value, $result);
     }
 
+    private function encodeAsSearch(OperatorInterface $value): stdClass
+    {
+        assert($value instanceof SearchStage || $value instanceof SearchMetaStage);
```

Comment:
Is there a reason these methods use `assert()` instead of strict type hints?

Reply:
Method removed.